packages/api/src/util/filterEvents.ts

```diff
@@ -7,24 +7,21 @@ import { l } from './logging';
 
 export function filterEvents (extHash: H256, { block: { extrinsics, header } }: SignedBlock, allEvents: EventRecord[], status: ExtrinsicStatus): EventRecord[] | undefined {
   // extrinsics to hashes
-  const myHash = extHash.toHex();
-  const allHashes = extrinsics.map((ext): string => ext.hash.toHex());
-
-  // find the index of our extrinsic in the block
-  const index = allHashes.indexOf(myHash);
-
-  // if we do get the block after finalized, it _should_ be there
-  if (index === -1) {
-    // only warn on filtering with isInBlock (finalization finalizes after)
-    if (status.isInBlock) {
-      l.warn(`block ${header.hash.toHex()}: Unable to find extrinsic ${myHash} inside ${allHashes.join(', ')}`);
+  for (const [index, extrinsic] of extrinsics.entries()) {
+    if (extHash === extrinsic.hash) {
```

Comment:
```suggestion
    if (extrinsic.hash.eq(extHash) {
```

The objects may not be the same, so we compare the raw bytes - will bail on the first mismatch.